files/manifest.go

```diff
@@ -0,0 +1,134 @@
+package files
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/amp-labs/cli/openapi"
+	"sigs.k8s.io/yaml"
+)
+
+const manifestVersion = "1.0.0"
+
+var ErrMissingField = errors.New("missing field")
+
+func ParseManifest(yamlData []byte) (*openapi.Manifest, error) {
+	manifest := &openapi.Manifest{}
+
+	if err := yaml.Unmarshal(yamlData, manifest); err != nil {
+		return nil, fmt.Errorf("failed to parse yaml: %w", err)
+	}
+
+	return manifest, nil
+}
+
+func ValidateManifest(manifest *openapi.Manifest) error {
+	if manifest.SpecVersion != manifestVersion {
+		return fmt.Errorf("%w: invalid spec version: %s (only %s is supported)",
+			ErrBadManifest, manifest.SpecVersion, manifestVersion)
+	}
+
+	if len(manifest.Integrations) == 0 {
+		return fmt.Errorf("%w: no integrations found in manifest, please define at least one integration",
+			ErrBadManifest)
+	}
+
+	for idx, integ := range manifest.Integrations {
+		if err := ValidateIntegration(idx, integ); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func ValidateProxy(proxy *openapi.IntegrationProxy) error {
+	if proxy.Enabled == nil {
+		return fmt.Errorf("%w: the enabled field is required", ErrMissingField)
+	}
+
+	return nil
+}
+
+func ValidateRead(read *openapi.IntegrationRead) error {
+	if read.StandardObjects == nil {
```

Comment:
let's change all of these to be looking for `read.Objects` instead, since we are making the shift very soon.